cql3/prepare_context.hh

```diff
@@ -38,11 +38,12 @@ private:
     std::vector<shared_ptr<column_identifier>> _variable_names;
 
     // Keeps column_specification for every bind_index. column_specification describes the name and type of this variable.
-    // It's different from _target_columns because it describes the name of this variable, not the target column.
     std::vector<lw_shared_ptr<column_specification>> _variable_specs;
 
-    // The column to which bind variable with a given bind_index will be assigned to.
-    std::vector<lw_shared_ptr<column_specification>> _target_columns;
+    // For every expression like (<target> = <bind variable>), there's a pair of (bind_index, target column_specification) in _targets.
+    // Collecting all equalities of bind variables allows to determine which of the variables set the value of partition key columns.
+    // The driver needs this information in order to compute the partition token and send the request to the right node.
+    std::vector<std::pair<std::size_t, lw_shared_ptr<column_specification>>> _targets;
```

Comment:
It makes sense to merge _targets and _variable_specs into a single data structure, but better keep that refactoring out of this series.

Reply:
We could merge it all into something like:

```c++
struct bind_variable_info {
    lw_shared_ptr<column_identifier> name_opt;
    lw_shared_ptr<column_specification> spec;
    std::vector<lw_shared_ptr<column_specifiction>> _targets;
};

std::vector<bind_variable_info> bind_variables;
```

It would be a tad prettier, but it's not that different from what we have now. I'm fine with leaving it as is, it's good enough. ¯\_(ツ)_/¯